docs/guide/yaml/test/example_yaml/paralleltestcase-example.yaml

```diff
@@ -0,0 +1,46 @@
+brooklyn.catalog:
+  id: parallel-test
+  version: 1.0
+  itemType: template
+  iconUrl: http://tomcat.apache.org/images/tomcat.png
+  name: Parallel Tomcat and DB Example Test
+  license: Apache-2.0
+  item:
+    brooklyn.config:
+      simple.confg: someValue
+    services:
+    - type: org.apache.brooklyn.test.framework.TestCase
+      name: Parallel Testcase
+      brooklyn.children:
+      - type: org.apache.brooklyn.test.framework.TestCase
+        name: Parallel Testcase
```

Comment:
Switching to the parallel example mentioned [here](https://github.com/apache/incubator-brooklyn/pull/1031#discussion_r44918698) and rewording the structural test entity descriptions to be less prescriptive on their usage.